add-ons/FAIRFEM/FEMPIREobjFctn.m

```diff
@@ -121,18 +121,25 @@
     H  = dr'*d2psi*dr + d2S;
 else
     % derivatives rather explicit
-    dr = dres*dT;
-    dD = dD*dT;
-    dJ = reshape(Mesh.mfPi(reshape(dD',[],Mesh.dim),'C'),1,[]) + dS;
+    %P  = @(x) reshape(Mesh.mfPi(reshape(x,[],dim),'C'),[],1);
+    %dTcmod = P((sdiag(det)*dT)')' + sdiag(Tc)*dDet;
+    P = kron(speye(dim),Mesh.PC);

```

Comment:
this could be a costly operation (which will be done in each call!). In principle, P is only used once and we have matrixFree code for it. 

I asssume `dT` below is sparse matrix of size n by dim*n, where n=prod(m). See if, supplying 'matrixFree',1 in the interpolation would render this a dense n by dim array. Then you can apply P or P' to each column in a matrix free fashion. 

This should enable you to avoid the `sdiag` as well. Can you give this a try?